ChronoKeeper/KeeperIdCard.h

```diff
@@ -0,0 +1,95 @@
+#ifndef CHRONOLOG_KEEPER_ID_CARD_H
+#define CHRONOLOG_KEEPER_ID_CARD_H
+
+#include <arpa/inet.h>
+
+#include <ostream>
+
+// this class wrapps ChronoKeeper Process identification 
+// that will be used by all the ChronoLog Processes 
+// to both identofy the Keepr process and create RPC client channels 
+// to send the data to the Keeper Recording service
+
+namespace chronolog
+{
+
+// Keeper Process can be uniquely identified by the combination of
+// the host IP address + client_port
+
+typedef uint32_t        in_addr_t;
+typedef uint16_t        in_port_t;
+typedef std::pair <in_addr_t, in_port_t> KeeperProcessId;
+
+// KeeperGroup is the logical grouping of KeeperProcesses
+typedef uint64_t    KeeperGroupId;
+
+
+class KeeperIdCard
+{
+
+   uint64_t keeper_group_id;
+   uint32_t ip_addr; //IP address as uint32_t in host byte order
+   uint16_t port;    //port number as uint16_t in host byte order
+   uint16_t tl_provider_id; // id of thallium service provider
+
+public:
+
+
+  KeeperIdCard( uint64_t group_id = 0, uint32_t addr = 0, uint16_t a_port=0, uint16_t provider_id=0)
+	: keeper_group_id(group_id), ip_addr(addr), port(a_port),tl_provider_id(provider_id) 
+	{} 
+
+  KeeperIdCard( KeeperIdCard const& other)
+  	: keeper_group_id(other.getGroupId()), ip_addr(other.getIPaddr()), port(other.getPort()),tl_provider_id(other.getProviderId())
+  	{}
+
+  ~KeeperIdCard()=default;
+
+  uint64_t getGroupId() const { return keeper_group_id; }
+  uint32_t getIPaddr() const {return ip_addr; }
+  uint16_t getPort() const { return port;}
+  uint16_t getProviderId () const { return tl_provider_id; }
+
+
+  // serialization function used by thallium RPC providers 
+  // to serialize/deserialize KeeperIdCard 
+  template <typename SerArchiveT>
+  void serialize( SerArchiveT & serT)
+  {
+      serT & keeper_group_id;
+      serT & ip_addr;
+      serT & port;	
+      serT & tl_provider_id;
```

Comment:
Follow the programming style used by the rest of ChronoLog, in the way we define RPCs, macros and prefixes used, binding functions etc. You have to define RPCs on the keeper that can be called by client processes. You should also define RPCs on the Visor that the keeper can call to update the KeeperTable and verify clients.

Reply:
@aparnasasidharan2017 . The code I submitted for review only claims to have initial implementation of Keeper - Visor communication,  Services on both Sides of this communication , KeeperRegistry , & Basic KeeperRecording Service with Ingestion.  Please, defer all your questions about the data repository organization for the future discussion.
Integration into the KeeperRegistry and addition of the Recording functionality into the CLient API were not part of this task either, as the existing interfaces need to be gradually refined as part of the integration process.

As to the programming style I suggest we have a discussion on the merits of both styles before we decide what style ChronoLog would follow going forward. We are very early in the development process to enforce a particular style